localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -646,9 +675,13 @@ void NDTScanMatcher::callbackSensorPoints(
   sensor_points_mapTF_msg.header.stamp = sensor_ros_time;
   sensor_points_mapTF_msg.header.frame_id = map_frame_;
   sensor_aligned_pose_pub_->publish(sensor_points_mapTF_msg);
+}
 
-  initial_pose_with_covariance_pub_->publish(initial_pose_cov_msg);
-
+void NDTScanMatcher::publish_marker(
+  const rclcpp::Time & sensor_ros_time,
+  const std::vector<Eigen::Matrix4f, Eigen::aligned_allocator<Eigen::Matrix4f>> &
```

Comment:
I think that it is easier to handle `std::vector<geometry_msgs::msg::Pose>` 
 instead of `std::vector<Eigen::Matrix4f, Eigen::aligned_allocator<Eigen::Matrix4f>>` .

To do that, it is better to create the following as the other function.
```
  for (const auto & pose_matrix : result_pose_matrix_array) {
    Eigen::Affine3d pose_affine;
    pose_affine.matrix() = pose_matrix.cast<double>();
    const geometry_msgs::msg::Pose pose_msg = tf2::toMsg(pose_affine);
    }
```

Reply:
Fixed in https://github.com/autowarefoundation/autoware.universe/pull/1757/commits/b255b43cde6c73522ef6cd4a66bd2af58cdffedb